lib/ssl/src/ssl_certificate.erl

```diff
@@ -589,18 +588,33 @@ verify_cert_extensions(Cert, #{cert_ext := CertExts} =  UserState) ->
 
 verify_cert_extensions(Cert, UserState, [], _) ->
     {valid, UserState#{issuer => Cert}};
-verify_cert_extensions(Cert, #{ocsp_responder_certs := ResponderCerts,
-                               ocsp_state := OscpState,
-                               issuer := Issuer} = UserState,
-                       [#certificate_status{response = OcspResponsDer} | Exts],
+verify_cert_extensions(Cert, #{stapling_state := StaplingState,
+                               issuer := Issuer,
+                               certdb := CertDbHandle,
+                               certdb_ref := CertDbRef} = UserState,
+                       [#certificate_status{response = OcspResponseDer} | Exts],
                        Context) ->
-    #{ocsp_nonce := Nonce} = OscpState,
-    case public_key:pkix_ocsp_validate(Cert, Issuer, OcspResponsDer,
-                                       ResponderCerts, Nonce) of
-        valid ->
+    #{ocsp_nonce := Nonce} = StaplingState,
+    IsTrustedResponderFun =
+        fun(#cert{der = DerResponderCert, otp = OtpCert}) ->
+                OtpTbsCert = OtpCert#'OTPCertificate'.tbsCertificate,
+                #'OTPTBSCertificate'{
+                   issuer = IssuerId, serialNumber = SerialNr} = OtpTbsCert,
+                case ssl_manager:lookup_trusted_cert(
+                       CertDbHandle, CertDbRef, SerialNr, IssuerId) of
+                    {ok, #cert{der = DerResponderCert}} ->
+                        true;
+                    _ ->
+                        false
+                end
+        end,
+    case public_key:pkix_ocsp_validate
+        (Cert, Issuer, OcspResponseDer, Nonce,
```

Comment:
Personally I dislike the newline between function name and left parenthesis.

Reply:
I would also prefer to brake it differently!  

```
case public_key:pkix_ocsp_validate(Cert, Issuer, 
                                                        OcspResponseDer, Nonce, 
                                                        ....                                        ) of 
```

Reply:
I also dislike it ;-)
'(' acts as operator here and it would be better if it does not start the new line.

I will fix it.

Reply:
fixed in 509784a4e6